gremlin-go/driver/performance/performanceSuite.go

```diff
@@ -0,0 +1,416 @@
+		var q *performanceResults
+		q, err = executeConnectionPooling(testName, g)
+		if err != nil {
+			return
+		}
+		performStats.executeDurationArr[i] = q.executeDuration.Seconds()
+		performStats.executeAllocsArr[i] = q.executeAllocs
+		performStats.executeBytesArr[i] = q.executeBytes
+	}
+	handleMetrics(performStats, "connection pooling", runs, memProf)
+}
+
+func handleMetrics(stats *performanceStats, testType string, runs int, memProf bool) {
+	fmt.Println("Results for", testType)
+	sort.Float64s(stats.executeDurationArr)
+	// treat first and last stat as outliers and remove
+	executeDurationArr := stats.executeDurationArr[:runs-1]
+	var totalTime float64
+	for i := range executeDurationArr {
+		totalTime += executeDurationArr[i]
+	}
+	printDuration(executeDurationArr, totalTime, runs-1)
+
+	if memProf {
+		sort.Slice(stats.executeAllocsArr, func(i, j int) bool { return stats.executeAllocsArr[i] < stats.executeAllocsArr[j] })
+		executeAllocsArr := stats.executeAllocsArr[:runs-1]
+		var totalAlloc uint64
+		for i := range executeAllocsArr {
+			totalAlloc += executeAllocsArr[i]
+		}
+		printAllocs(executeAllocsArr, totalAlloc, runs-1)
+
+		sort.Slice(stats.executeBytesArr, func(i, j int) bool { return stats.executeBytesArr[i] < stats.executeBytesArr[j] })
+		executeBytesArr := stats.executeBytesArr[:runs-1]
+		var totalBytes uint64
+		for i := range executeBytesArr {
+			totalBytes += executeBytesArr[i]
+		}
+		printBytes(executeBytesArr, totalBytes, runs-1)
+	}
+}
+
+func printDuration(stats []float64, total float64, runs int) {
+	fmt.Printf("\tEXECUTION STATS:\taverage %fs\tmedian %fs\tp90 %fs \tp95 %fs\tmax %fs\tmin %fs\n",
+		total/float64(runs), stats[int(math.Ceil(float64(runs/2)))],
+		stats[int(math.Ceil(float64(runs*90/100)))-1], stats[int(math.Ceil(float64(runs*95/100)))-1], stats[runs-1], stats[0])
+}
+
+func printAllocs(stats []uint64, total uint64, runs int) {
+	fmt.Printf("\tALLOCATION STATS:\taverage %d allocs\tmedian %d allocs\tp90 %d allocs\tp95 %d allocs\tmax %d allocs\tmin %d allocs\n",
+		total/uint64(runs), stats[runs/2],
+		stats[int(math.Ceil(float64(runs*90/100)))-1], stats[int(math.Ceil(float64(runs*95/100)))-1], stats[runs-1], stats[0])
+}
+
+func printBytes(stats []uint64, total uint64, runs int) {
+	fmt.Printf("\tMEMORY USE STATS:\taverage %d B\tmedian %d B\tp90 %d B\tp95 %d B\tmax %d B\tmin %d B\n",
+		total/uint64(runs), stats[runs/2],
+		stats[int(math.Ceil(float64(runs*90/100)))-1], stats[int(math.Ceil(float64(runs*95/100)))-1], stats[runs-1], stats[0])
+}
+
+func getTraversal(testName string, g *GraphTraversalSource) *GraphTraversal {
+	if traversalFcn, ok := traversalMap[testName]; ok {
+		return traversalFcn(g)
+	} else {
+		return nil
+	}
+}
+
+// connection setting constants
+
+const Host = "localhost"
+const Port = 45940
+const GremlinWarning = gremlingo.Warning
+const gratefulGraphAlias = "ggrateful"
+const threshold = 4 // same as default
+const writeBufferSize = 1073741824
```

Comment:
can we use buffer size same as in Java driver?

Reply:
Oh, it was the same (math.MaxInt32), since Java also needed to increase their content size. Should I ask Rithin to change his Java size to this one instead?

Reply:
@lyndonbauto are buffer size in gremlin-go and content size in java driver the same?

Reply:
No I don't think so, would suggest making them the same @xiazcy @vkagamlyk

Reply:
Sorry misunderstodd, thought you were asking if they are the same value, yes I think they are the same @vkagamlyk

Reply:
so we should have default buffer size 64k?
https://github.com/lyndonbauto/tinkerpop/blob/gremlin-go-ms6/gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Connection.java#L64